pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,365 @@
+        } else {
+            // Growing unrecognized content, nothing to do.
+            return Promise.resolve();
+        }
+    }
+
+    function crypto_resize() {
+        if (crypto) {
+            const opts = { };
+            if (passphrase)
+                opts.passphrase = { t: "s", v: passphrase };
+            return crypto.Resize(size - crypto_overhead, opts);
+        } else {
+            return Promise.resolve();
+        }
+    }
+
+    function lvm_resize() {
+        if (size != lvol.Size)
+            return lvol.Resize(size, { });
+        else
+            return Promise.resolve();
+    }
+
+    if (size < orig_size) {
+        return fsys_resize().then(crypto_resize)
+                .then(lvm_resize);
+    } else if (size >= orig_size) {
+        return lvm_resize().then(crypto_resize)
+                .then(fsys_resize);
+    }
+}
+
+export function get_resize_info(client, block, to_fit) {
+    let info, shrink_excuse, grow_excuse;
+
+    if (block) {
+        if (block.IdUsage == 'crypto' && client.blocks_crypto[block.path]) {
+            const cleartext = client.blocks_cleartext[block.path];
+
+            if (!cleartext) {
+                info = { };
+                shrink_excuse = grow_excuse = _("Encrypted volumes need to be unlocked before they can be resized.");
+            } else {
+                return get_resize_info(client, cleartext, to_fit);
+            }
+        } else if (block.IdUsage == 'filesystem') {
+            info = client.fsys_info && client.fsys_info[block.IdType];
+
+            if (!info) {
+                info = { };
+                shrink_excuse = grow_excuse = cockpit.format(_("$0 filesystems can not be resized here."),
+                                                             block.IdType);
+            } else {
+                if (!info.can_shrink)
+                    shrink_excuse = cockpit.format(_("$0 filesystems can not be made smaller."),
+                                                   block.IdType);
+                if (!info.can_grow)
+                    grow_excuse = cockpit.format(_("$0 filesystems can not be made larger."),
+                                                 block.IdType);
+            }
+        } else if (client.blocks_stratis_blockdev[block.path] && client.features.stratis_grow_blockdevs) {
+            info = {
+                can_shrink: false,
+                can_grow: true,
+                grow_needs_unmount: false
+            };
+            shrink_excuse = _("Stratis blockdevs can not be made smaller");
+        } else if (block.IdUsage == 'raid') {
+            info = { };
+            shrink_excuse = grow_excuse = _("Physical volumes can not be resized here.");
+        } else if (client.legacy_vdo_overlay.find_by_backing_block(block)) {
+            info = {
+                can_shrink: false,
+                can_grow: true,
+                grow_needs_unmount: false
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19400-20230927-121722-6ccb869e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)